src/parser/esprima_cpp/esprima.cpp

```diff
@@ -1386,9 +1392,11 @@ class Parser {
     template <class ASTBuilder>
     void parseFormalParameters(ASTBuilder& builder, ParseFormalParametersResult& options, Scanner::SmallScannerResult* firstRestricted = nullptr, bool allowTrailingComma = true)
     {
-        // parseFormalParameters only scans the parameter list of function
-        // should be invoked only during the global parsing (program)
-        ASSERT(!builder.isNodeGenerator() && !this->isParsingSingleFunction);
+// parseFormalParameters only scans the parameter list of function
+// should be invoked only during the global parsing (program)
+#if defined(ESCARGOT_SMALL_CONFIG)
+        ASSERT(!builder.willGenerateByteCode() && !this->isParsingSingleFunction);
+#endif
```

Comment:
It seems that this ASSERT statement should be checked when it is not `SMALL_CONFIG` mode.

Reply:
IMO this ASSERT can be applied regardless of `SMALL_CONFIG` mode. 
What do you think about it?

Reply:
you are right